src/keywords/optionalint.cpp

```diff
@@ -55,21 +56,32 @@ std::string OptionalIntegerKeyword::textWhenNull() const { return textWhenNull_;
 // Deserialise from supplied LineParser, starting at given argument offset
 bool OptionalIntegerKeyword::deserialise(LineParser &parser, int startArg, const CoreData &coreData)
 {
-    if (parser.hasArg(startArg))
+    std::optional<int> newValue;
+    try
     {
-        auto x = parser.argd(startArg);
-        if (!setData(x))
+        newValue = parser.argi(startArg);
+        if (parser.hasArg(startArg))
         {
-            if (maximumLimit_)
-                Messenger::error("Value {} is out of range for keyword. Valid range is {} <= n.\n", x, maximumLimit_.value());
-
-            return false;
+            auto x = parser.argi(startArg);
+            if (!setData(x))
+            {
+                if (maximumLimit_)
+                    Messenger::error("Value {} is out of range for keyword. Valid range is {} <= n.\n", x,
```

Comment:
I should have added the name of the keyword to the error message:
```suggestion
                    Messenger::error("Value {} is out of range for keyword '{}'. Valid range is {} <= n.\n", x, name_,
```